### PR TITLE
Feature/interval indices

### PR DESCRIPTION
**Pull request by kfwilson**
this is a cleaned up version of the previous work I did to add intervals to the indices. I want these changes in first so I can build off them for the wide on draws things without having the first PR for that be huge. I tested this building an artifact - it's slow but it works.

### CODE REVIEW

File: src/vivarium_inputs/data_artifact/builder.py [hunk 1]
```diff
@@ -101,6 +101,9 @@ def process(self, entity_key: EntityKey) -> None:
 def _worker(entity_key: EntityKey, location: str, modeled_causes: Collection[str],
             artifact: Artifact) -> None:
     data = loader(entity_key, location, modeled_causes, all_measures=False)
+    # FIXME: This is a hack since hdf files can't handle pandas.Interval objects
+    data = split_interval('age', 'age_group', data)
```
Comment (collijk) [inline src/vivarium_inputs/data_artifact/builder.py]: Reorder the arguments here and use keywords to clarify what's happening.  E.g. Calls should look like
```
data = split_interval(data, interval_column='age', split_column_prefix='age_group')
```

File: src/vivarium_inputs/utilities.py [hunk 2]
```diff
@@ -42,18 +42,18 @@ def scrub_sex(data):
 def scrub_age(data):
     if 'age_group_id' in data.index.names:
         age_bins = utility_data.get_age_bins().set_index('age_group_id')
-        starts = list(data.index.levels[data.index.names.index('age_group_id')].map(age_bins['age_group_start']))
-        data = (data.assign(age_group_end=(data.index.get_level_values('age_group_id')
-                                           .map(age_bins['age_group_end'])))
-                .set_index('age_group_end', append=True))
-        data.index = data.index.rename('age_group_start', 'age_group_id').set_levels(starts, 'age_group_start')
+        id_levels = data.index.levels[data.index.names.index('age_group_id')]
+        interval_levels = [pd.Interval(age_bins.age_group_start[id],
```
Comment (collijk) [inline src/vivarium_inputs/utilities.py]: ``id`` is a builtin you should avoid using as a variable name.

Comment (kfwilson) [reply src/vivarium_inputs/utilities.py]: changed

File: src/vivarium_inputs/data_artifact/builder.py (hunk 1 above)
Comment (kfwilson) [reply src/vivarium_inputs/data_artifact/builder.py]: done

**Pull request merged**